### PR TITLE
MueLu: region MG Chebyshev updated to match ifpack2

### PR DESCRIPTION
**Pull request by pohm01**
@trilinos/muelu 

## Motivation

An update to the region MG Chebyshev relaxation so that it more closely resembles the implementation in Ifpack2.

mayrmt: added command line argument to control boost factor. Included this also into tests.

## Testing

Matches Ifpack2 on single region for unsmoothed aggregation.

mayrmt: ran `ctest -R MueLu_Structured_Region` and `ctest -R UnitTestsRegion` locally. All passed.

### CODE REVIEW

File: packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionSmoothers_def.hpp [hunk 1]
```diff
@@ -413,16 +413,19 @@ void chebyshevIterate ( RCP<Teuchos::ParameterList> params,
   const int maxIter      = params->get<int>   ("smoother: sweeps");
   const Scalar eigRatio  = params->get<double>("smoother: eigRatio");
   const Scalar lambdaMax = params->get<Scalar>("chebyshev: lambda max");
-  const Scalar lambdaMin = lambdaMax / eigRatio;
+  const Scalar boostFactor = 1.1; // TODO: add as a param
```
Comment (mayrmt) [inline packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionSmoothers_def.hpp]: @pohm01 You could read this parameter from the command line in the driver and then pass it thought the parameter list as entry `chebyshev: boost factor`.

Comment (mayrmt) [reply packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionSmoothers_def.hpp]: Added the command line argument `--smootherChebyBoostFactor` to control this parameter from the command line.

**Comment by mayrmt** [comment]
@pohm01 I added the command line argument to control the boost factor and took the liberty to update the PR description accordingly.

**Comment by mhoemmen** [comment]
Yay code duplication?  I guess MueLu can't depend directly on Ifpack2, but maybe it would make sense for MueLu to have a single common interface to algorithms like Chebyshev.

**Pull request merged**